src/sagemaker_xgboost_container/constants/xgb_constants.py

```diff
@@ -16,6 +16,7 @@
     'auc',
     'aucpr',
     'f1',
+    'f1macro',
```

Comment:
Not completely sure about the string. Where I can see the possible metrics strings to be used for AutoPilot?

Reply:
It is used in hyperparameter settings, e.g. eval_metric or tuning metric, in format `f1` or `validation:f1`. XGB will strip off `validation:` part.

Reply:
Whenever this fix is accepted and merged, we should also update Autopilot pipeline recommender. In the file `algorithm/algo_metadata/config/sagemaker-xgboost/v0/sagemaker-xgboost.json`, update `F1` metric for multiclass to `validation:f1macro`. Thanks.

Reply:
> It is used in hyperparameter settings, e.g. eval_metric or tuning metric, in format `f1` or `validation:f1`. XGB will strip off `validation:` part.

Thanks Haifeng. In pipeline recommender (algorithm/core/const.py), the metrics are `F1` and `F1_MACRO`, is it somewhere in this package, all the metrics names are converted to lower cases such that we could use 'f1' here? If this is true, should I use `f1_macro` instead of 'f1macro'?

Reply:
> Whenever this fix is accepted and merged, we should also update Autopilot pipeline recommender. In the file `algorithm/algo_metadata/config/sagemaker-xgboost/v0/sagemaker-xgboost.json`, update `F1` metric for multiclass to `validation:f1macro`. Thanks.

I think we use accuracy as default metric for multi-class datasets in AutoPilot. User can override this choice to `f1macro` though.

Reply:
The metric name conversion is defined in `algorithm/algo_metadata/config/sagemaker-xgboost/v0/sagemaker-xgboost.json`. It is up to you to define it as `f1macro` or `f1_macro`.

Note that we define multiple metrics for each problem type. We have F1 and ACCURACY defined for multiclass. ACCURACY is the default one. It doesn't mean that you cannot define other metrics for multiclass.

Reply:
I suggest `f1_macro` - easier to parse

Reply:
Alright, thanks. Will change it to `f1_macro`.

Reply:
@iyerr3 @lihaife How about XGBoost 1.0? Do we need to change it there as well?

Reply:
I don't suggest back port breaking changes.